### PR TITLE
games-emulation/*: update JGEMU packages

### PR DESCRIPTION
**Pull request by orbea**
* Added new `USE` flags for bsnes-jg: `doc`, `examples`, `jgmodule`, and `shared`
* Version bump for jgrf, bsnes-jg and sameboy-jg
* Drop old bsnes-jg and sameboy-jg versions

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: games-emulation/bsnes-jg/bsnes-jg-2.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+fi
+
+LICENSE="
+	ISC GPL-3+ LGPL-2.1+ MIT ZLIB
+	examples? ( 0BSD )
+"
+SLOT="1"
+IUSE="examples +jgmodule shared"
+REQUIRED_USE="
+	|| ( examples jgmodule shared )
```
Comment (juippis) [inline games-emulation/bsnes-jg/bsnes-jg-2.0.0.ebuild:31]: Eh, I guess it compiles/installs nothing if none of these are set? :)

**Review by juippis — APPROVED** [review]
bsnes-jg are broken with: `* USE:        abi_x86_64 amd64 elibc_glibc examples kernel_linux`
```
>>> Install games-emulation/bsnes-jg-2.0.0 into /var/tmp/portage/games-emulation/bsnes-jg-2.0.0/image
make -j32 install DESTDIR=/var/tmp/portage/games-emulation/bsnes-jg-2.0.0/image DOCDIR=/usr/share/doc/bsnes-jg-2.0.0 LIBDIR=/usr/lib64 PREFIX=/usr DISABLE_MODULE=1 ENABLE_EXAMPLE=1 ENABLE_SHARED=0 
cp bsnes/boards.bml /var/tmp/portage/games-emulation/bsnes-jg-2.0.0/image/usr/share/jollygood/bsnes/
cp objs/lib/example/bsnes-example /var/tmp/portage/games-emulation/bsnes-jg-2.0.0/image/usr/bin/
for i in COPYING README; do \
    cp /var/tmp/portage/games-emulation/bsnes-jg-2.0.0/work/bsnes-2.0.0/$i \
        /var/tmp/portage/games-emulation/bsnes-jg-2.0.0/image/usr/share/doc/bsnes-jg-2.0.0; \
done
cp: cannot stat 'bsnes/boards.bml': No such file or directory
make: *** [Makefile:239: install-data] Error 1
make: *** Waiting for unfinished jobs....
 * ERROR: games-emulation/bsnes-jg-2.0.0::gentoo failed (install phase):
 *   emake failed
 * 
```
I guess `examples` expect `+doc` to be set? This isn't too critical for me, I'd like to merge this PR without prolonging it, so if you could investigate and fix this afterwards it'd be great.

**Comment by orbea** [comment]
@juippis Thank you for the review and merge! I made a new PR to address the issue you pointed out. https://github.com/gentoo/gentoo/pull/39186

I fixed it upstream by fixing the path of the asset (.bml) files.

File: games-emulation/bsnes-jg/bsnes-jg-2.0.0.ebuild (hunk 1 above)
Comment (orbea) [reply games-emulation/bsnes-jg/bsnes-jg-2.0.0.ebuild:31]: Yep, the Makefile would print `make: Nothing to be done for 'all'.` in that case.

**Pull request closed without merging**